### PR TITLE
GF-56 - Authentication with devise token auth

### PR DESCRIPTION
**Pull request by MIKALAKE**

### CODE REVIEW

File: app/controllers/api_controller.rb [hunk 1]
```diff
@@ -1,3 +1,9 @@
 class ApiController < UnauthenticatedApiController
-  @current_user = nil
+  before_action :authenticate_v1_user!
+
+  protected
+
+  def current_user
+    @current_user = current_v1_user
+  end
 end
```
Comment (iamdavidpetric) [inline app/controllers/api_controller.rb:9]: empty line bellow

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,2 +1,17 @@
...
+  before_action :configure_permitted_parameters, if: :devise_controller?
+
+  before_action do
+    request.session_options[:skip] = true
+  end
+
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up, keys: %i[first_name last_name email password avatar_url])
+  end
 end
```
Comment (iamdavidpetric) [inline app/controllers/application_controller.rb:17]: empty line

File: app/models/user.rb [hunk 3]
```diff
@@ -12,6 +12,11 @@
 #  updated_at :datetime         not null
 #
 class User < ApplicationRecord
+  # Include default devise modules.
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable,
+         :confirmable
```
Comment (iamdavidpetric) [inline app/models/user.rb]: make it one line

File: app/controllers/api_controller.rb (hunk 1 above)
Comment (MIKALAKE) [reply app/controllers/api_controller.rb:9]: done

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (MIKALAKE) [reply app/controllers/application_controller.rb:17]: done

File: app/models/user.rb (hunk 3 above)
Comment (MIKALAKE) [reply app/models/user.rb]: done

File: app/controllers/v1/sponsors_controller.rb [hunk 4]
```diff
@@ -1,5 +1,6 @@
 module V1
   class SponsorsController < ApiController
```
Comment (iamalextodoran) [inline app/controllers/v1/sponsors_controller.rb]: Are you actually using this controller?

File: app/models/user.rb [hunk 5]
```diff
@@ -12,6 +12,9 @@
 #  updated_at :datetime         not null
 #
 class User < ApplicationRecord
+  # Include default devise modules.
+  devise :database_authenticatable, :registerable, :recoverable, :rememberable, :validatable, :confirmable
+  include DeviseTokenAuth::Concerns::User
```
Comment (iamalextodoran) [inline app/models/user.rb:15]: Move this up, above the commented code.

File: app/views/devise/sessions/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,67 @@
+<!doctype html>
```
Comment (iamalextodoran) [inline app/views/devise/sessions/new.html.erb]: Delete this file

File: app/views/devise/sessions/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+
+
+
+
+
+
+
+
+Message good-fellas-2022
```
Comment (iamalextodoran) [inline app/views/devise/sessions/new.html.erb]: 👀

File: config/initializers/cors.rb [hunk 8]
```diff
@@ -1,4 +1,4 @@
-# Be sure to restart your server when you modify this file.
+3 # Be sure to restart your server when you modify this file.
```
Comment (iamalextodoran) [inline config/initializers/cors.rb]: 👀

File: config/locales/devise.en.yml [hunk 9]
```diff
@@ -0,0 +1,7 @@
+en:
```
Comment (iamalextodoran) [inline config/locales/devise.en.yml]: Delete this file
@iamdavidpetric what's up doc?

File: config/locales/devise.ja.yml [hunk 10]
```diff
@@ -0,0 +1,7 @@
+en:
```
Comment (iamalextodoran) [inline config/locales/devise.ja.yml]: Delete this file
@iamdavidpetric what's up doc?

File: config/locales/ja.yml [hunk 11]
```diff
@@ -0,0 +1,33 @@
+# Files in the config/locales directory are used for internationalization
```
Comment (iamalextodoran) [inline config/locales/ja.yml]: Delete this file
@iamdavidpetric what's up doc?

File: config/routes.rb [hunk 12]
```diff
@@ -1,9 +1,10 @@
 Rails.application.routes.draw do
   namespace :v1 do
+    mount_devise_token_auth_for "User", at: "auth"
+    resources :sessions, only: [:create, :destroy]
```
Comment (iamalextodoran) [inline config/routes.rb]: There's no sessions controller therefore not needed.

File: db/seeds.rb [hunk 13]
```diff
@@ -6,7 +6,7 @@
 #   movies = Movie.create([{ name: "Star Wars" }, { name: "Lord of the Rings" }])
 #   Character.create(name: "Luke", movie: movies.first)
 
-user = User.create({ first_name: "Paul", last_name: "Pop", avatar_url: "https://upload.wikimedia.org/wikipedia/commons/6/62/CEVA_Logistics_New_Logo.png", email: "paulpop122@gmail.com", admin: true, gender: "male" })
+# user = User.create({ first_name: "Paul", last_name: "Pop", avatar_url: "https://upload.wikimedia.org/wikipedia/commons/6/62/CEVA_Logistics_New_Logo.png", email: "paulpop122@gmail.com", admin: true, gender: "male" })
```
Comment (iamalextodoran) [inline db/seeds.rb]: Fix this user creation in here.

**Review by iamalextodoran — CHANGES_REQUESTED** [review]
@iamdavidpetric

File: app/controllers/v1/sponsors_controller.rb (hunk 4 above)
Comment (MIKALAKE) [reply app/controllers/v1/sponsors_controller.rb]: done

File: app/models/user.rb (hunk 5 above)
Comment (MIKALAKE) [reply app/models/user.rb:15]: done

Comment (MIKALAKE) [reply app/models/user.rb:15]: done

File: app/views/devise/sessions/new.html.erb (hunk 6 above)
Comment (MIKALAKE) [reply app/views/devise/sessions/new.html.erb]: done

File: config/routes.rb (hunk 12 above)
Comment (MIKALAKE) [reply config/routes.rb]: done

File: db/seeds.rb (hunk 13 above)
Comment (MIKALAKE) [reply db/seeds.rb]: done

File: config/locales/devise.ja.yml (hunk 10 above)
Comment (MIKALAKE) [reply config/locales/devise.ja.yml]: done

File: config/locales/devise.en.yml (hunk 9 above)
Comment (MIKALAKE) [reply config/locales/devise.en.yml]: done

File: config/locales/ja.yml (hunk 11 above)
Comment (MIKALAKE) [reply config/locales/ja.yml]: done

File: config/initializers/cors.rb (hunk 8 above)
Comment (MIKALAKE) [reply config/initializers/cors.rb]: done

File: app/controllers/v1/sponsors_controller.rb (hunk 4 above)
Comment (MIKALAKE) [reply app/controllers/v1/sponsors_controller.rb]: done

File: db/seeds.rb [hunk 14]
```diff
@@ -6,7 +6,7 @@
 #   movies = Movie.create([{ name: "Star Wars" }, { name: "Lord of the Rings" }])
 #   Character.create(name: "Luke", movie: movies.first)
 
-user = User.create({ first_name: "Paul", last_name: "Pop", avatar_url: "https://upload.wikimedia.org/wikipedia/commons/6/62/CEVA_Logistics_New_Logo.png", email: "paulpop122@gmail.com", admin: true, gender: "male" })
+# user = User.create({ first_name: "Paul", last_name: "Pop", avatar_url: "https://upload.wikimedia.org/wikipedia/commons/6/62/CEVA_Logistics_New_Logo.png", email: "paulpop122@gmail.com", password:'lamanele123' admin: true})
```
Comment (iamalextodoran) [inline db/seeds.rb]: Uncomment and try it, see if it works.
`rails db:seed`

Comment (MIKALAKE) [reply db/seeds.rb]: No it does not work

Comment (MIKALAKE) [reply db/seeds.rb]: What can I change MASTER @iamalextodoran ?

Comment (iamalextodoran) [reply db/seeds.rb]: Follow the path of errors my young padawan

**Review by iamalextodoran — APPROVED** [review]

**Review by iamalextodoran — APPROVED** [review]

**Pull request merged**